unit-tests/src/test/scala/java/nio/file/FilesSuite.scala

```diff
@@ -891,6 +894,64 @@ object FilesSuite extends tests.Suite {
     }
   }
 
+  // This test was inspired by Issue #1354
+  // "NIO File Walker fails on broken links."
+  test("Files.walkFileTree respects FileVisitOption.FOLLOW_LINKS") {
+    withTemporaryDirectory { dirFile =>
```

Comment:
Outside of the scope of this PR, but it feels like there should be a nio TestHelpers that provides a `withTemporaryDirectory` method that provides a `Path` rather than a `File` so that we don't have a mismatch of io and nio apis in the test suites.

Reply:
I can take a run at creating the helper you describe.  It would make things easier for folks,
including myself, to follow in the future.

Reply:
I took the first step towards following the suggestion for a TestHelpers.scala by
creating and using a method withTemporaryDirectoryPath().   I hope that 
method captures the concept. At the least, I provides an implementation for 
review and continuous improvement.

I used the new method in the code I was adding only and did not do sweeping 
changes to many other places which used withTemporaryDirectory().

If/when useTemporaryDirectoryPath() passes review, I can create an Issue to
factor withTemporaryDirectory(), withTemporaryDirectory(), and the core routine
they both use, into a separate file.  That probably means changing the imports
for several other nio Suites which currently use withTemporaryDirectory().

?? Should the new file be called withTemporary.scala so that future devos can
easily find the file containing the methods?  I guess that a grep of TestHelpers.scala
would work also.

Thoughts?